### PR TITLE
Global opt testing range

### PR DESCRIPTION
**Pull request by maxsiska**

### CODE REVIEW

File: notebooks/Example_Fitting_a_Linear_Model.ipynb [hunk 1]
```diff
@@ -39,14 +39,12 @@
    "outputs": [
```
Comment (michaelosthege) [inline notebooks/Example_Fitting_a_Linear_Model.ipynb:39]: <p>Please put the comments into the text cell above</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/JuBiotech/calibr8/pull/54/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (michaelosthege) [inline notebooks/Example_Fitting_a_Linear_Model.ipynb:39]: <p>Please pass <code>ci_prob=0.9</code> =) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/JuBiotech/calibr8/pull/54/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

**Comment by michaelosthege** [comment]
Please also bump the version in the 2nd digit while you're at it.

File: calibr8/optimization.py [hunk 2]
```diff
@@ -156,6 +156,104 @@ def fit_scipy(
...
+    independent : array-like
+        desired values of the independent variable or measured values of the same
+    dependent : array-like
+        observations of dependent variable
+    theta_bounds : array-like
+        bounds to fit the parameters
+    method: str, optional
+        Type of solver. Should be
+            - 'dual_annealing'         : ref:`(see here) <optimize.dual_annealing>`
+        If not given, chosen to be ``dual_annealing``.
```
Comment (lhelleckes) [inline calibr8/optimization.py:184]: Why making this optional is there is only one option? Do you want to keep the method flesxibel for the future?
For now, this kwarg is unnecessary.

File: calibr8/tests.py [hunk 3]
```diff
@@ -1285,6 +1285,58 @@ def test_fit_pygmo(self, caplog):
         numpy.testing.assert_array_equal(em.cal_dependent, y)
         pass
 
+    def test_fit_scipy_global(self, caplog):
+        numpy.random.seed(2345)
```
Comment (lhelleckes) [inline calibr8/tests.py:1289]: Where are you using this seed? It's better not to set it gloablly (you could forget to reset it, but rather instantiate a random number generator with the seed and use this for all further options):

`rng = rng = numpy.random.RandomState(seed)` and later sth like `rng.permutation` or `rng.uniform`

**Review by lhelleckes — COMMENTED** [review]
Nice work, a few minor things

File: notebooks/Example_Fitting_a_Linear_Model.ipynb (hunk 1 above)
Comment (lhelleckes) [inline notebooks/Example_Fitting_a_Linear_Model.ipynb:39]: <p>The arguments should be <code>hdi_lower </code>and <code>hdi_upper </code>here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/JuBiotech/calibr8/pull/54/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

File: calibr8/optimization.py (hunk 2 above)
Comment (maxsiska) [reply calibr8/optimization.py:184]: There was support for multiple solvers just two commits ago. However, I couldn't produce a scenario where these alternative solvers (scipy global opt alternatives) outperformed dual_annealing.
That does not mean that such a scenario is not possible.
Either way, global optimization is a particular topic. I plead for expanding functionality by providing different solvers that users can choose from at their discretion.

**Comment by lhelleckes** [comment]
@m.osthege The comment on `numpy.random.seed` in the tests in a more general discussion. I am not sure why we have it there and if we should avoid setting the seed this way. I would say so, but please comment as well.

**Comment by michaelosthege** [comment]
> @michaelosthege The comment on `numpy.random.seed` in the tests in a more general discussion. I am not sure why we have it there and if we should avoid setting the seed this way. I would say so, but please comment as well.

IMO it's not super dramatic that we use `np.random.seed` here.
If you want to fix this, we'll need to pass `random_state=rng` in the `scipy.stats....rvs()` call in `_get_test_model`.

**Review by michaelosthege — APPROVED** [review]
Let's move ahead with this 👍

We fixed the docstring stuff locally and now the rendered docs look good too.

**Pull request merged**